phylanx/config.hpp

```diff
@@ -11,8 +11,30 @@
 #include <phylanx/config/export_definitions.hpp>
 #include <phylanx/config/version.hpp>
 
+// TODO: If windows.h is included, ensure min and max macros are not defined
+// after here or the code will fail to compile pointing at header files in
+// Blaze. Therefore, the ordering of the includes matter. This needs a
+// permanent fix
 #if defined(_MSC_VER) && !defined(NOMINMAX)
 #define NOMINMAX
-#endif
+#endif // NOMINMAX
+
+// Use HPX threads. Do not use C++11 threads
+#if !defined(BLAZE_USE_HPX_THREADS)
+#define BLAZE_USE_HPX_THREADS 1
+
+#if !defined(BLAZE_USE_SHARED_MEMORY_PARALLELIZATION) || BLAZE_USE_SHARED_MEMORY_PARALLELIZATION != 1
+#define BLAZE_USE_SHARED_MEMORY_PARALLELIZATION 1
```

Comment:
This will cause problems if `BLAZE_USE_SHARED_MEMORY_PARALLELIZATION == 0` or similar.

Reply:
If `BLAZE_USE_SHARED_MEMORY_PARALLELIZATION ` is not 1 then it may not use HPX. Why is being equal to 0 a problem?

Reply:
if  `BLAZE_USE_SHARED_MEMORY_PARALLELIZATION` is defined but `!= 1` you will see macro redefinition errors from the preprocessor.